Run a Network Server (#186)
helium/docs: docs/use-the-network/run-a-network-server/buy-an-oui/buy-an-oui.mdx

---
id: buy-an-oui
hide_title: true
sidebar_label: Buy an OUI
slug: /use-the-network/buy-an-oui
---

import useBaseUrl from "@docusaurus/useBaseUrl";

# What is an OUI?

Each LoRaWAN Network Server (LNS) on the Helium Network acquires an
Organizationally Unique Identifier (OUI). This registers the LNS with the
blockchain and allocates two very important routing objects owned and maintained
by the OUI operator:

- up to five **filters**, describing the sets (AppEUI, DevEUI) expected by the
  LNS

> - one or more **slabs** of NwkAddrs, reserving an address space for the OUI's

Reviewer: Should this be dev address?

Author: I think they are pretty interchangeable, but I do think calling it a "DevAddr" slab might be easier since it's the higher level identifier. I'll correct the [LoRaWAN section](https://docs.helium.com/lorawan-on-helium) to be consistent about "slab" indications.

  devices on the Helium Network

# OUI Cost

> The OUI is purchased with data credits (DC). Costs are subject to change, but

Reviewer 2: ```suggestion
The OUI is purchased with [data credits (DC)](/blockchain/transaction-fees). Costs are subject to change, but
```

currently the OUI itself costs $100 worth of DCs and each DevAddr cost an
additional $100 in DC.

DevAddr are sold in sequential blocks between 8 and 65,536 and any power of two.
You must purchase a slab when purchasing an OUI, therefore, the minimum cost for

> an OUI is $800 for 8 DevAddr's and $100 for the OUI itself.

Reviewer 2: ```suggestion
an OUI is $800 for eight DevAddr's and $100 for the OUI itself.
```

[The OUI purchase transaction itself](https://github.com/helium/proto/blob/master/src/blockchain_txn_oui_v1.proto)
incurs a fee based as does any other blockchain transaction; generally this will

> be $0.35-1.00 worth of DC based on how the fields get filled and change the

Reviewer 2: ```suggestion
be $0.35-1.00 worth of DCs based on how the fields get filled and change the
```

transaction size.

# Purchase Transaction

You need [the Helium wallet](https://github.com/helium/helium-wallet-rs) to go
any further. You'll want to
[create a wallet](https://github.com/helium/helium-wallet-rs#create-a-wallet)

> and to have a bit over $900 in HNT.

Reviewer 2: ```suggestion
and to have at least $900 in HNT (base on current HNT Oracle pricing).
```

## Create Data Credits

Many fees benefit from

> [implicit burn](http://localhost:3000/blockchain/transaction-fees#transaction-fees-and-implicit-burn),

Reviewer: localhost link?

Author: Thanks - good catch

but when paying for OUIs and DevAddr's, the Data Credits must be part of the
purchasers balance.

```text
$ ./helium-wallet balance
+-----------------------------------------------------+------------+--------------+-----------------+
| Address                                             | Balance    | Data Credits | Security Tokens |
+-----------------------------------------------------+------------+--------------+-----------------+
| 12yUUZmsVyVaomxkvuNXtD2rAHYWgYwJLXtz4H9FjYgE2WcniLx | 0.00000000 | 0            | 0.00000000      |
+-----------------------------------------------------+------------+--------------+-----------------+
```

> If we don't have an account balance of at least 90000000 DCs ($900), we need to

Reviewer 2: ```suggestion
If we don't have an account balance of at least `90,000,000` DCs ($900), we need to
```

create the data credits now.

Inspect
[the current oracle price](https://api.helium.io/v1/oracle/prices/current) and
you'll get the current price of HNT as agreed by the Helium Network; it's
represented in 10^-8 dollars. Suppose it is currently $4.5, so we would get:

```text
{"data":{"timestamp":"2021-02-25T10:01:32.000000Z","price":450000000,"block":735970}}
```

Since 1 DC = $.00001 = $10^-5 and we need to normalize with the $10^-8 from the

> oracle price, we can plug the following into our favorite calculate (Python for

Reviewer 2: ```suggestion
oracle price, we can plug the following into our favorite calculate (for example, in Python):
```

> me):

Reviewer 2: ```suggestion
```

| Address                                             | Balance    | Data Credits | Security Tokens |
+-----------------------------------------------------+------------+--------------+-----------------+
| 12yUUZmsVyVaomxkvuNXtD2rAHYWgYwJLXtz4H9FjYgE2WcniLx | 0.00000000 | 90225000     | 0.00000000      |
+-----------------------------------------------------+------------+--------------+-----------------+
```

## Purchase the OUI

The next part is simple. Submit a "create OUI" transaction:

```text
./helium-wallet oui create --subnet-size 8 --filter wVwCiewtCpEKAAAAAAAAAAAAcCK3fwAAAAAAAAAAAABI7IQOAHAAAAAAAAAAAAAAAQAAADBlAAAAAAAAAAAAADEAAAA2AAAAOgAAAA
```

The filter is a dummy filter to initialize the OUI. When you get Console
running, it will automatically maintain it. You can once again track the
transaction with the endpoint above. When it is complete, you should also see
your OUI as the [most recently purchased](https://api.helium.io/v1/ouis/last)
and on
[this list here](https://api.helium.io/v1/ouis?cursor=eyJ0eXBlcyI6Im91aV92MSIsImJsb2NrIjo3MDUxMDAsImFuY2hvcl9ibG9jayI6NzA1MTAwf).

Congratulations! You are the proud owner of a Helium Network OUI. OUIs are

> numbered sequentially, so you get street cred for small numbers.

Reviewer 2: ```suggestion
numbered sequentially, so the lower you are, the earlier you were on the Network!.
```

Author: Shame. I thought you'd enjoy discussions of street cred

Reviewer 2: I'm a fan, but I'm not so sure about non-native English speakers.

Author: yea I agree.
